client/pagination.go

```diff
@@ -0,0 +1,67 @@
+package client
+
+import "strconv"
+
+const limit = 100
+
+type Pagination struct {
+	offset int
+	params map[string]string
+}
+
+// Note: all the FIXME should be uncommented when Alpine adds support for go 1.18.
+// This will enable generics.
+
+type Paginated interface {
+	// FIXME: Environment
```

Comment:
Once generic is used, we can add all the "elements" that support pagination.
E.g. `Environment | Module | Project`